geps/gep-91/index.md

```diff
@@ -1,22 +1,135 @@
 
 ## Non-Goals
-- Define other fields that can be used to verify the client certificate such as the Certificate Hash or Subject Alt Name. 
+- Define other fields that can be used to verify the client certificate such as the Certificate Hash. 
+
+## Existing support in Implementations
+
+This feature is widely supported in implementations that support Gateway API.
+This table highlights the support. Please feel free to add any missing implementations not mentioned below.
+
+| Implementation | Support       |
+|----------------|------------|
+| Apache APISIX  | [ApisixTls.Client.CASecret](https://apisix.apache.org/docs/ingress-controller/tutorials/mtls/#mutual-authentication)      |
+| Contour        | [HTTPProxy.Spec.VirtualHost.Tls.ClientValidation.CASecret](https://projectcontour.io/docs/v1.17.1/config/tls-termination/)      |
+| Emissary Ingress| [TlSContext.Spec.Secret](https://www.getambassador.io/docs/emissary/latest/topics/running/tls/mtls)     |
+| Gloo Edge      | [VirtualService.Spec.SSLConfig.SecretRef](https://docs.solo.io/gloo-edge/latest/guides/security/tls/server_tls/#configuring-downstream-mtls-in-a-virtual-service)      |
+| Istio          | [Gateway.Spec.Servers.TLS.Mode](https://istio.io/latest/docs/tasks/traffic-management/ingress/secure-ingress/#configure-a-mutual-tls-ingress-gateway)      |
+| Kong           | [mTLS Plugin](https://docs.konghq.com/hub/kong-inc/mtls-auth/)      |
+| Traefik        | [TLSOption.Spec.ClientAuth](https://doc.traefik.io/traefik/https/tls/#client-authentication-mtls)    |
+
+### API
+
+* Introduce a `Client` field of type `TLSClientContext` within [GatewayTLSConfig][] to hold TLS configuration specific to the client.
+* Introduce a `Validation` field of type `TLSValidationContext` within the `Client` field that can be used to validate the peer with which the TLS connection is being made.
+to the Gateway.
+* Introduce a `caCertificateRefs` field within `ValidationContext` that can be used to specify a list of CA Certificates that
+can be used as a trust anchor to validate the certificates presented by the client.
+* This new field is mutually exclusive with the [BackendTLSPolicy][] configuation which is used to validate the TLS certificate presented by the peer on the connection between the Gateway and the backend, and this GEP is adding support for validating the TLS certificate presented by the peer on the connection between the Gateway and the downstream client.
+* Introduce an optional `subjectAltNames` field within `ClientValidationContext` that can be used to specify one or more alternate names to verify the subject identity in the certificate presented by the client. The maximum number of alternate names that can be specified is implementation defined.
+
+#### GO
+
+```go
+// TLSClientContext holds configuration specific to the client initiating the TLS connection.
+type TLSClientContext struct {
```

Comment:
Worth mentioning that adding this field will result in servers adding a 'certificate request' in the TLS server hello - which results in browsers asking for a cert in a dialog. This is pretty unfortunate UX for the browsers - many users are not aware of this. 

So best to use a separate hostname or port for the mtls if browsers are used.

Reply:
@costinm is ^ the case even if `CACertificateRefs` refs a CA cert that resides as an anchor in the client's system trust bundle?

Reply:
Yes. 

mTLS works by server asking the client for a cert in ServerHelo - if server doesn't ask, client won't send. And if server asks, a browser is required to show the cert selection dialog ( at least first time ? Haven't tried in a long time). 

Note that most servers have options to make client cert optional or required - both still ask for cert, but one will allow the request to proceed and use JWT or other forms of auth if client cert is not available. 

This should probably be covered in the 'prior art' and background section...

Reply:
thanks ! incorporated this in as a doc string